Added and implemented a yoda gopher.

schwerdt opened the pull request:
Added and implemented a yoda gopher.
The yoda gopher API only accepts 5 requests per hour. Use
then wisely.

Implemented the ChuckNorris gopher.
Implemented the Emoji gopher for 2 emojis.

schwerdt committed:
Added and implemented a yoda gopher.

The yoda gopher API only accepts 5 requests per hour. Use
then wisely.

Implemented the ChuckNorris gopher.
Implemented the Emoji gopher for 2 emojis.

carolynvs commented on gophers/chuck_norris_gopher.go:
```diff
@@ -11,13 +18,35 @@ func NewChuckNorrisGopher() ChuckNorrisGopher {
+  }
+
+  // Need to take the body of the response and turn into a byte array
+  body, err := ioutil.ReadAll(response.Body)
```
You can combine reading the bytes with unmarshaling, saves you a bit of extra error handling.

```golang
err = json.NewDecoder(resp.Body).Decode(&api_struct)
```

schwerdt committed:
Simplified the decoding of API response for Chuck Norris Gopher.

Simplified the structure into which the json is read so that
I do not have to declare a separate structure for 'Value'.  Instead
I have a nested structure for the API response.

schwerdt committed:
Adding a test for yoda gopher!
